helpers/lib/resourceHints.js

```diff
@@ -41,22 +48,30 @@ function addResourceHint(globals, path, state, type) {
         return;
     }
 
-    let value = {src: path, state};
+    let hint = {src: path, state};
 
-    if (typeof type !== 'string') {
-        type = '';
+    if (utils.isString(type) && resourceHintTypes.includes(type)) {
+        hint.type = type;
     }
-    type = type.trim();
-    if (type !== '' && resourceHintAllowedTypes.includes(type)) {
-        value.type = type;
+
+    if (!utils.isString(cors) || !allowedCors.includes(cors)) {
```

Comment:
don't you think it's better to throw an error if we use some random cors value, rather then fallback to "no"?

Reply:
you are right. we should default to cors only when no value is provided as that is the default a link tag and link headers have.